java-checks/src/test/files/checks/RedundantThrowsDeclaration.java

```diff
@@ -287,3 +287,47 @@ public void missingJavadocForException() throws MyException, MyException2, java.
 abstract class Test {
   abstract void foo() throws Unknown, Unknown;
 }
+
+public class Parent {
+  public Parent() throws IllegalAccessException { // Compliant
+    throw new IllegalAccessException();
+  }
+
+  public Parent(String a) {
+
+  }
+
+  public Parent(String a, String b) throws IllegalAccessException { // Compliant
+    foo();
+  }
+
+  private void foo() throws IllegalAccessException {
+    throw new IllegalAccessException();
+  }
+}
+
+public class Child extends Parent {
+
+  public Child(Integer a) throws IllegalAccessException { // Compliant, implicit call can throw IllegalAccessException
+    // implicit call to parent constructor
+    System.out.println("a:" + a);
+  }
+
+  public Child(Long a) throws IllegalAccessException { // Compliant
+    super();
+    // no implicit call to parent constructor
```

Comment:
being completely explicit, I think you can remove the comments here and in the following methods